src/engine/MaxConstraint.cpp

```diff
@@ -406,8 +374,14 @@ bool MaxConstraint::satisfied() const
     if ( !( _assignment.exists( _f ) && _assignment.size() > 0 ) )
```

Comment:
I guess we really want every assignment to be present because I'm not sure we can safely conclude the constraint is satisfied otherwise. Updated to check this.